### DISCUSSION TITLE
posts/optimization-in-situ-merge-sort-o-1-the-space-complexity/

### DISCUSSION THREAD

**Discussion by giscus** [pluveto/blog-comments#6] [Announcements]
# posts/optimization-in-situ-merge-sort-o-1-the-space-complexity/

https://www.less-bug.com/posts/optimization-in-situ-merge-sort-o-1-the-space-complexity/

**Comment by codingknees** [comment]
归并排序和快排的时间复杂度都是O(nlgn)，且归并也可以in-place实现，为什么各大排序库中，快排更加受欢迎呢？

**Reply by YHX5129** [reply]
快速排序常数因子小，最快，大O时间复杂度略去了常数因子
